quickwit-search/src/search_stream/collector.rs

```diff
@@ -194,6 +189,264 @@ impl FastFieldCollectorBuilder {
     }
 }
 
+#[derive(Clone)]
+pub struct PartionnedFastFieldCollectorBuilder {
+    fast_field_value_type: Type,
+    fast_field_name: String,
+    partition_by_fast_field_value_type: Type,
+    partition_by_fast_field_name: String,
+    timestamp_field_name: Option<String>,
+    timestamp_field: Option<Field>,
+    start_timestamp: Option<i64>,
+    end_timestamp: Option<i64>,
+}
+
+impl PartionnedFastFieldCollectorBuilder {
+    pub fn new(
+        fast_field_value_type: Type,
+        fast_field_name: String,
+        timestamp_field_name: Option<String>,
+        timestamp_field: Option<Field>,
+        start_timestamp: Option<i64>,
+        end_timestamp: Option<i64>,
+        partition_by_fast_field_value_type: Type,
+        partition_by_fast_field_name: String,
+    ) -> crate::Result<Self> {
+        match fast_field_value_type {
+            Type::U64 | Type::I64 => (),
+            _ => {
+                return Err(SearchError::InvalidQuery(format!(
+                    "Fast field type `{:?}` not supported",
+                    fast_field_value_type
+                )));
+            }
+        }
+        Ok(Self {
+            fast_field_value_type,
+            fast_field_name,
+            partition_by_fast_field_value_type,
+            partition_by_fast_field_name,
+            timestamp_field_name,
+            timestamp_field,
+            start_timestamp,
+            end_timestamp,
+        })
+    }
+
+    pub fn value_type(&self) -> (Type, Type) {
+        (
+            self.fast_field_value_type,
+            self.partition_by_fast_field_value_type,
+        )
+    }
+
+    pub fn fast_field_to_warm(&self) -> HashSet<String> {
+        let mut set = HashSet::new();
+        set.insert(self.fast_field_name.clone());
+        if let Some(timestamp_field_name) = &self.timestamp_field_name {
+            set.insert(timestamp_field_name.clone());
+            set.insert(self.partition_by_fast_field_name.clone());
+        }
+        set
+    }
+
+    pub fn typed_build<TFastValue: FastValue, TPartitionValue: FastValue>(
+        &self,
+    ) -> PartionnedFastFieldCollector<TFastValue, TPartitionValue> {
+        PartionnedFastFieldCollector::<TFastValue, TPartitionValue> {
+            fast_field_to_collect: self.fast_field_name.clone(),
+            partition_by_fast_field: self.partition_by_fast_field_name.clone(),
+            timestamp_field_opt: self.timestamp_field,
+            start_timestamp_opt: self.start_timestamp,
+            end_timestamp_opt: self.end_timestamp,
+            _marker: PhantomData,
```

Comment:
just for aesthetic purpose: we can use only one marker with a pair `PhantomData<(Item, PartitionItem)>`